jina/flow/base.py

```diff
@@ -393,6 +398,76 @@ def gather_inspect(
             # no inspect node is in the graph, return the current graph
             return self
 
+    def _get_gateway_target(self, prefix):
+        gateway_pod = self._pod_nodes[GATEWAY_NAME]
+        return (
+            f'{prefix}-{GATEWAY_NAME}',
+            {
+                'pod_address': gateway_pod.full_address,
+                'expected_parts': 0,
+            },
+        )
+
+    def _update_port_type(self, pod):
+        pod.head_args.socket_in = SocketType.ROUTER_BIND
+        pod.tail_args.dynamic_out_routing = True
+        pod.tail_args.socket_out = SocketType.DEALER_CONNECT
+
+    def _get_routing_graph(self):
```

Comment:
I think this can be `unified` with the code that is done now in `Flow.build` where a `map` from starting Pods to outgoing Pods is built by name.

This could be the input of this way of building the graph

Reply:
No the other can just be removed. Not needed anymore. As I wrote in slack: cleanup is pending...